libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -2736,24 +2737,60 @@ def build_peripheral_list(self):
             done.add(type)
         return peripherals
 
+    def get_processed_defaults_file(self, defaults_filepath, depth=0):
+        '''reads defaults_filepath, expanding any @include lines to include
+        the contents of the so-references file - recursively.'''
+        if depth > 10:
+            raise Exception("include loop")
+        ret = ""
+        with open(defaults_filepath, 'r') as defaults_fh:
+            while True:
+                line = defaults_fh.readline()
+                if line == "":
+                    break
+                m = re.match("^\s*@include\s*([^\s]+)", line)
```

Comment:
why allow white space at start of the line? I think it should be hard against the margin

Reply:
Changed